### ISSUE TITLE
Emscripten: could not compile with --preload-file bin/data@data

### ISSUE THREAD

**Issue by artificiel** [openframeworks/openFrameworks#7593]
i have an app with an empty data dir that would not compile with `--preload-file bin/data@data`

```
em++: error: '..../emsdk/upstream/emscripten/tools/file_packager bin/randomexplorer.data --from-emcc --preload bin/data@data' failed (returned 1)
make[1]: *** [bin/randomexplorer.html] Error 1
make: *** [Release] Error 2
emmake: error: 'make' failed (returned 2)
```

not sure how to debug that or what is relevant? before the above lines the output it pretty OF standard and just some warnings, mostly macro redefinitions.

taking out the `--preload-file bin/data@data` makes the compile complete.

**Comment by Jonathhhan** [comment]
@artificiel the alternative is to create an empty data folder (most examples without data have an data folder with a `.gitkeep` file).

**Comment by artificiel** [comment]
ah! but I had an empty bin/data -- but with really nothing, no .gitkeep. 

adding a .gitkeep made the compile work.

I would expect the compile to work with a truly empty bin/data? and/or have better errors? without a bin/data we get `error: bin/data@data does not exist` which is clear and actionable. I now deduce that the logging for a truly empty data dir is `Nothing to do!` which, if it turns out to be an error, should be more explicit.

but better: absent or empty bin/data should not hinder the process.

**Comment by Jonathhhan** [comment]
> but better: absent or empty bin/data should not hinder the process.
I agree, but I guess thats an Emscripten issue...

**Comment by artificiel** [comment]
Ah ok — I’m very thin on emscripten and how it works (what it does vs how to use it) I was under the impression that bin/data was a specific OF thing that could be conditionally tested (I.e. if bin/data does not exist or is empty, do not preload-file).

could the of emscripten OF guide be amended with the strict requirement of having a non-empty bin/data directory? (Because I understood the first message and created the bin/data, but the second problem threw me off, and since it’s my first emscripten use, it’s not so clear how things  are supposed to happen)

**Comment by Jonathhhan** [comment]
Maybe it is possible to only use the flag, if there is a data folder with some content?

**Comment by artificiel** [comment]
Yeah that would work — not sure how/where to implement… I’ve edited the title of this issue to make it more like a “todo” element.

**Comment by Jonathhhan** [comment]
@artificiel also not sure how to implement it, but I guess this is the place:
https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworksCompiled/project/emscripten/config.emscripten.default.mk
